Documentation/ABI/testing/sysfs-kernel-error_report

```diff
@@ -0,0 +1,46 @@
+What:		/sys/kernel/error_report/
+Date:		January 2021
+Contact:	Alexander Potapenko <glider@google.com>,
+		Marco Elver <elver@google.com>
+Description:
+		/sys/kernel/error_report/ contains two files: "report_count"
+		and "last_report". These files are used to notify the userspace
+		about the error reports from the enrolled kernel subsystems
+		(those that use error_report_start/error_report_end
+		tracepoints).
+
+		"report_count" contains the current number of reported errors.
+		This number is incremented every time the error_report_end
+		trace event occurs in the kernel.
+
+		"last_report" contains one most recent (on the best-effort
+		basis; the last report that started without any other report
```

Comment:
But I'm still a little confused by the second part. Perhaps just:
```
concurrent error reports results in any one report being collected.
```

So the final sentence here could just be:
```
"last_report" contains the most recent error report; concurrent report generation results in collection of any one report ("last_report" may not be the last shown on the console).
```

I'm not sure if best-effort applies, because the caveat is about concurrency. printk() ultimately serializes messages, and there is some order, but if we end up concurrently generating reports, we just can't predict which report will end up being collected by looking at the console. The implementation resolves the race by picking the first one to acquire current_pid, but that may not necessarily be the one that appears before the other concurrent reports (e.g. if it just stalls for a while).

So for all intents and purposes, from a behaviour POV, choosing any one of the concurrent reports is perfectly reasonable. Our implementation currently satisfies that, but I don't think we need to be more precise in case the implementation changes in future.

Reply:
Good point, I don't think the exact details of which of the concurrent reports is picked are important. I took your version.